src/number/format.js

```diff
@@ -79,6 +105,22 @@ return function( number, properties ) {
 			minimumFractionDigits, maximumFractionDigits, round, roundIncrement );
 	}
 
+	// Compact mode: apply formatting
+	if ( compactMap && compactPattern ) {
+		pluralForm = pluralGenerator ? pluralGenerator( number ) : "other";
+		compactPattern = compactMap[ "1" + zeroes + "-count-" + pluralForm ] || compactPattern;
+
+		// Some languages specify no pattern for certain digit lengths, represented as "0".
+		// Only apply compact pattern if one is specified.
+		if ( compactPattern !== "0" ) {
```

Comment:
Above, could you simply set `compactPattern = null` in this case and avoid this if completely?